src/WaveFunction/WaveFunction.cpp

```diff
@@ -62,4 +63,188 @@ double WaveFunction::calculateShannonEntropy() const {
 }
 
 
+/**
+ *  Transform the underlying ONV basis of the wave function (only for FCI [ProductFockSpace]) as shown in Helgaker 11.9
+ *
+ *  @param T    the transformation matrix between the old and the new orbital basis
+ */
+void WaveFunction::basisTransform(const SquareMatrix<double>& T) {
+
+    if (fock_space->get_type() != FockSpaceType::ProductFockSpace) {
+        throw std::invalid_argument("WaveFunction::basisTransform(SquareMatrix<double>): This is not an FCI wave function");
+    }
+
+    auto K = fock_space->get_K();
+
+    if (K != T.get_dim()) {
+        throw std::invalid_argument("WaveFunction::basisTransform(SquareMatrix<double>): number of orbitals does not match the dimension of the transformation matrix T");
+    }
+
+    // Retrieve LU decomposition for T
+    const auto& LU = T.NoPivotLUDecomposition();
+
+    // Retrieve L
+    SquareMatrix<double> L = SquareMatrix<double>::Identity(K, K);
+    L.triangularView<Eigen::StrictlyLower>() = LU[0];
+
+    // Retrive U
+    SquareMatrix<double> U = SquareMatrix<double>(LU[1].triangularView<Eigen::Upper>());
+
+
+    SquareMatrix<double> U_in = U.inverse();
+
+    // Calculate t (the operator which allows per-orbital transformation of the wave function)
+    SquareMatrix<double> t =  SquareMatrix<double>::Identity(K, K) - L + U_in;
+
+    // FockSpace
+    const auto& product_fock_space = dynamic_cast<const ProductFockSpace&>(*fock_space);
+    const FockSpace& fock_space_alpha = product_fock_space.get_fock_space_alpha();
+    const FockSpace& fock_space_beta = product_fock_space.get_fock_space_beta();
+
+    auto dim_alpha = fock_space_alpha.get_dimension();
+    auto dim_beta = fock_space_beta.get_dimension();
+    auto N_alpha = fock_space_alpha.get_N();
+    auto N_beta = fock_space_beta.get_N();
+
+    // Wave function transformtion algorithm
+    VectorX<double> current_coefficients = this->coefficients;  // coefficients will be updated after each orbital transform (C^(n-1)) in Helgaker
+    // each iteration will calculate a set of correction coefficients (Delta C in Helgaker) to update "current_coefficients".
```

Comment:
align this with the other documentation comment, or better: do the explaining of what is going to happen after line 109.